Cache-Hash-Data-Structure.adoc

```diff
@@ -0,0 +1,174 @@
+= Hash Data Structure
+
+Provide a Hash data structure to Infinispan users. The Hash is a flexible identifiable grouping of keys mapping to values.
+In this context, _flexible_ means it can have arbitrary mappings without needing a schema beforehand; _identifiable group_
+means the Hash structure receives a unique name and retrieves the complete key-value mapping.
+
+The Hash provides a general approach for users to represent their data structures using Infinispan. For example, by
+employing this, users can avoid using key prefixes to create an object in Cache.
+
+== Public API
+
+The work is for the new API only. The Hash data structure has the format of:
+
+[source,java]
+.Hash.java
+----
+public interface Hash<K, V> { // <1>
```

Comment:
How is this different from cache ?

Reply:
Because it is stored as a single entry in a cache.

Reply:
ok, now I get it

Reply:
We could build upon MultiMap without duplicates, storing key-value pairs. But I am unsure if we can compare for equality based only on the key. On the other hand, maybe we could abstract some of the MultiMap implementations which work over a Collection to work on another structure instead of creating everything from scratch. I would need to look at that, though.

Reply:
isn't it the same as `FineGrainedAtomicMap` that we dropped and replaced with the `Grouping API`?